src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -147,7 +148,10 @@ public static MetadataReference SystemCoreRef
             {
                 if (s_systemCoreRef == null)
                 {
-                    s_systemCoreRef = AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.System_Core).GetReference(display: "System.Core.v4_0_30319.dll");
+                    Interlocked.CompareExchange(
+                        ref s_systemCoreRef,
+                        AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.System_Core).GetReference(display: "System.Core.v4_0_30319.dll"),
```

Comment:
It would seem we would always assign this to the same value. 
Do we care about reference identity of the MetadataReference? If yes, then we probably should fix other lazy initializers as well.